apps/sample-app/src/App.tsx

```diff
@@ -0,0 +1,68 @@
+import './App.css';
+
+import { Button, FluentProvider, Spinner, teamsLightTheme, Text, Theme } from '@fluentui/react-components';
+import { app, authentication } from '@microsoft/teams-js';
+import React, { useState } from 'react';
+
+import { ProfileContent } from './components/Profile';
+import { appInitializationFailed, getTheme } from './components/utils';
+
+const App: React.FC = () => {
+  const [isInitialized, setIsInitialized] = React.useState(false);
+  const [accessToken, setAccessToken] = React.useState<string>();
+
+  React.useEffect(() => {
+    (async () => {
+      await app.initialize();
+      setIsInitialized(true);
+    })();
+  }, [setIsInitialized]);
+
+  const handle = React.useCallback(async () => {
+    const result = await authentication.authenticate({
+      url: 'https://localhost:4003/?auth=1',
+    });
+    setAccessToken(result);
+  }, [setAccessToken]);
+
+  const [currTheme, setCurrTheme] = useState<Theme>(teamsLightTheme);
+
+  React.useEffect(() => {
```

Comment:
I don't think React guarantees the order in which `useEffect` callbacks are executed. So I'm wondering if there's a potential for this one to run before the one on line 14 is done, which would result in API calls being made before the app is properly initialized. 

Can you move all this to a single useEffect instead?